app/src/lib/components/CommitDragItem.svelte

```diff
@@ -1,131 +1,60 @@
-		if (!$branch) return;
-
-		return (data: any) => {
-			if (data instanceof DraggableHunk) {
-				const newOwnership = `${data.hunk.filePath}:${data.hunk.id}`;
-				branchController.amendBranch($branch.id, commit.id, newOwnership);
-			} else if (data instanceof DraggableFile) {
-				if (data.file instanceof LocalFile) {
-					// this is an uncommitted file change being amended to a previous commit
-					const newOwnership = filesToOwnership(data.files);
-					branchController.amendBranch($branch.id, commit.id, newOwnership);
-				} else if (data.file instanceof RemoteFile) {
-					// this is a file from a commit, rather than an uncommitted file
-					const newOwnership = filesToSimpleOwnership(data.files);
-					if (data.commit) {
-						branchController.moveCommitFile($branch.id, data.commit.id, commit.id, newOwnership);
-					}
-				}
-			}
-		};
+	interface Props {
+		commit: Commit | RemoteCommit;
+		children: Snippet;
 	}
 
-	function acceptSquash(commit: Commit | RemoteCommit) {
-		if (!$branch) return;
+	const { commit, children }: Props = $props();
 
-		if (commit instanceof RemoteCommit) {
-			return () => false;
-		}
-		return (data: any) => {
-			if (!(data instanceof DraggableCommit)) return false;
-			if (data.branchId !== $branch.id) return false;
-
-			if (data.commit.isParentOf(commit)) {
-				if (data.commit.isIntegrated) return false;
-				if (data.commit.isRemote && !project.ok_with_force_push) return false;
-				return true;
-			} else if (commit.isParentOf(data.commit)) {
-				if (commit.isIntegrated) return false;
-				if (commit.isRemote && !project.ok_with_force_push) return false;
-				return true;
-			} else {
-				return false;
-			}
-		};
-	}
+	const branch = maybeGetContextStore(Branch);
 
-	function onSquash(commit: Commit | RemoteCommit) {
+	const actions = $derived.by(() => {
 		if (!$branch) return;
 
-		if (commit instanceof RemoteCommit) {
-			return () => false;
-		}
-		return (data: DraggableCommit) => {
-			if (data.commit.isParentOf(commit)) {
-				branchController.squashBranchCommit(data.branchId, commit.id);
-			} else if (commit.isParentOf(data.commit)) {
-				branchController.squashBranchCommit(data.branchId, data.commit.id);
-			}
-		};
-	}
+		return commitDragActionsFactory.build($branch, commit);
+	});
 </script>
 
 <div class="commit-list-item">
-	<div
-		class="commit-card-wrapper"
-		use:dropzone={{
-			active: 'amend-dz-active',
-			hover: 'amend-dz-hover',
-			accepts: acceptAmend(commit),
-			onDrop: onAmend(commit)
-		}}
-		use:dropzone={{
-			active: 'squash-dz-active',
-			hover: 'squash-dz-hover',
-			accepts: acceptSquash(commit),
-			onDrop: onSquash(commit)
-		}}
-	>
-		<!-- DROPZONES -->
-		<DropzoneOverlay class="amend-dz-marker" label="Amend" />
-		<DropzoneOverlay class="squash-dz-marker" label="Squash" />
-
-		<slot />
+	<div class="commit-card-wrapper">
+		{#if actions}
+			{@render ammendDropzone()}
+		{:else}
+			{@render children()}
+		{/if}
```

Comment:
This component needs to gracefully fall back to not talking to the branchController because its used in the remoteBranch and trunk pages.

I thought about having the snippets inside the if block too, but it ended up looking too messy